src/azul/chalice.py

```diff
@@ -40,6 +40,22 @@ def __init__(self, app_name, unit_test=False, spec=None):
             self._specs: Optional[MutableJSON] = None
         super().__init__(app_name, debug=config.debug > 0, configure_logs=False)
 
+    def _bearer_token(self, request: Request) -> Optional[str]:
+        try:
+            auth_header = request.headers.get('Authorization') or request.headers['authorization']
+        except KeyError:
+            return None
+        else:
+            try:
+                auth_type, auth_token = auth_header.split()
+            except ValueError:
+                return None
+            else:
+                if auth_type.lower() == 'bearer':
+                    return auth_token
+                else:
+                    return None
```

Comment:
We should reject requests carrying an authorization header that doesn't match our expectation. The response code should be 401.

Reply:
Raising 401 here causes a series of obtuse unit test failures stemming from here: https://github.com/DataBiosphere/azul/blob/dbee69ecf692c8268ab262b96b9d4b89f75595fb/test/indexer/test_hca_indexer.py#L1472

Seems that the indexer also uses the `Authorization` header for hmac.

Reply:
That may be so in which case we need to add the appropriate conditional behavior to accommodate all valid cases. The service doesn't use HMAC and so it should let that pass. Vice versa, the indexer doesn't use bearer tokens and shouldn't let that pass. When it comes to security, leniency is usually ill advised.